client/network_list.go

```diff
@@ -0,0 +1,22 @@
+package client
+
+import (
+	"context"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+// NetworkList lists all the networks.
+func (client *APIClient) NetworkList(ctx context.Context) (*types.NetworkListResp, error) {
+	resp, err := client.get(ctx, "/networks", nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	network := &types.NetworkListResp{}
```

Comment:
I do not think the naming of network makes sense.

Maybe `netListResp := &types.NetworkListResp{}` is a little better.